### DISCUSSION TITLE
How to remove span name in log message?

### DISCUSSION THREAD

**Discussion by notinmybackyaard** [tokio-rs/tracing#2424] [Q&A] [answered]
Hello!
Would you tell me how to setup tracing config in my situation?
I like to use tracing crate with log and flame feature in my program.
And I want to profile span events with flame, but don't want to print span name in log message..
Here's my code below

```
fn setup_fmt() -> impl Layer<Registry> {
    fmt::Layer::default()
        .with_target(false)
        .with_file(true)
        .with_line_number(true)
        .with_ansi(false)
        .with_span_events(fmt::format::FmtSpan::NONE)
        .with_filter(INFO)
}

fn setup_flame() -> (impl Layer<Registry>, FlushGuard<BufWriter<File>>) {
    let (layer, guard) =
        FlameLayer::with_file("./tracing.folded").unwrap();
    let filtered = layer
        .with_file_and_line(false)
        .with_filter(INFO);
    (filtered, guard)
}

fn main() {
    let fmt_layer = setup_fmt();
    let (flame_layer, _guard) = setup_flame();

    Registry::default()
        .with(fmt_layer.and_then(flame_layer))
        .init();

   foo();
}

fn foo() {
    let _span = info_span!("SpanForFlame");
    info!("Log")
}
```

Expected Log:
2022-12-23T04:08:31.298650Z INFO Log

Actual Log:
2022-12-23T04:08:31.298650Z INFO SpanForFlame: Log

**Comment by elkowar** [comment, marked as answer]
You can use a [filter](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/index.html#filtering-with-layers) on your fmt-layer, for example a [filter_fn](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/fn.filter_fn.html) which filters out span events for that layer.

I.e.:
```rust
tracing_subscriber::registry()
    .with(format_layer.with_filter(filter::filter_fn(move |meta| {
        !meta.is_span() || include_span_info
    })))
    .with(env_filter)
    .init();
```
